src/cylinder.jl

```diff
@@ -43,8 +43,9 @@ function endcircles(s::Cylinder{3})
     b2 = cross(b1, s.a)
     axes = [s.a[1] b1[1] b2[1]; s.a[2] b1[2] b2[2]; s.a[3] b1[3] b2[3]]
     d = 2*s.r
-    return(Ellipsoid(s.c + s.a*s.h2, SVector(0.0, d, d), axes),
-           Ellipsoid(s.c - s.a*s.h2, SVector(0.0, d, d), axes))
+    et = Ellipsoid(s.c + s.a*s.h2, SVector(0.0, d, d), axes)::Ellipsoid{3,Void,9}  # top disk
+    eb = Ellipsoid(s.c - s.a*s.h2, SVector(0.0, d, d), axes)::Ellipsoid{3,Void,9}  # bottom disk
```

Comment:
It turns out that all the outer constructors are type-unstable.  For example, even for the simplest `Box`, we get
```julia

julia> VERSION
v"0.5.3-pre+5"

julia> @code_warntype Box([0,0],[2,4])
Variables:
  #self#::Type{GeometryPrimitives.Box}
  c::Array{Int64,1}
  d::Array{Int64,1}

Body:
  begin
      SSAValue(1) = (Base.arraylen)(c::Array{Int64,1})::Int64
      SSAValue(0) = (Base.arraylen)(c::Array{Int64,1})::Int64
      return $(Expr(:invoke, LambdaInfo for GeometryPrimitives.Box{N,D,L}(::Array{Int64,1}, ::Array{Int64,1}, ::Array{Float64,2}, ::Void), :(#self#), :(c), :(d), :($(Expr(:invoke, LambdaInfo for eye(::Type{T}, ::Int64, ::Int64), :(Base.eye), :(Base.Float64), SSAValue(1), SSAValue(0)))), :(GeometryPrimitives.nothing)))
  end::GeometryPrimitives.Box{N,D,L}
```
(Note that the type parameters `N`, `D`, `L` are not determined.)  This instability occurs because the outer constructor for `Box` returns `Box{length(c),typeof(data),length(c)^2}(...)`, whose type parameter `length(c)` cannot be inferred.

Similarly, the `Ellipsoid(...)` calls at the end of the `endcircles` are unstable, leading to instability of `endcircles`.

My solution to remove this instability without type assertion is to add another outer constructor taking `SVector` and `SMatrix`, from which the values of `N` and `L` can be inferred.  For example, define
```julia
function Ellipsoid{N,D,L,R<:Real}(c::SVector{N,R}, d::SVector{N,R}, axes::SMatrix{N,N,R,L},
                   data::D=nothing)
    return Ellipsoid{N,D,L}(c, inv(axes ./ sqrt.(sum(abs2,axes,1))), (d*0.5) .^ -2, data)
end
```
I just checked that calling this constructor inside `endcircles` solved the problem.

Any better suggestions?